test/distributed/checkpoint/test_dtensor_checkpoint.py

```diff
@@ -79,9 +85,32 @@ def create_dtensor_model(
         replicated_dt = distribute_tensor(
             tensor_to_replicate, mesh, placements=[Replicate()]
         )
-        model = MyTestModule(sharded_dt, replicated_dt).cuda(dist.get_rank())
 
-        return model, sharded_dt, replicated_dt
+        # Only even rank will be part of the mesh.
+        submesh = DeviceMesh(
+            device_type=self.device_type,
+            mesh=[i for i in range(dist.get_world_size()) if i % 2 == 0],
+        )
+        submesh_sharded_dt = distribute_tensor(
+            tensor_to_shard_to_submesh, submesh, placements=[Shard(0)]
+        )
+        submesh_replicated_dt = distribute_tensor(
+            tensor_to_replicate_submesh, submesh, placements=[Replicate()]
+        )
+        model = MyTestModule(
+            sharded_dt,
+            replicated_dt,
+            submesh_sharded_dt,
+            submesh_replicated_dt,
+        ).cuda(dist.get_rank())
```

Comment:
nit: we don't need to put the model on cuda again, as we already put individual parameters to their own device in dtensor construction